src/guacscanner/guacscanner.py

```diff
@@ -0,0 +1,758 @@
+            Optional("--vpc-id"): Or(
+                None,
+                And(
+                    str,
+                    Use(str.lower),
+                    lambda x: VPC_ID_REGEX.match(x) is not None,
+                    error="Possible values for --vpc-id are the characters vpc- followed by either 8 or 17 hexadecimal digits.",
+                ),
+            ),
+            str: object,  # Don't care about other keys, if any
+        }
+    )
+    try:
+        validated_args = schema.validate(args)
+    except SchemaError as err:
+        # Exit because one or more of the arguments were invalid
+        print(err, file=sys.stderr)
+        sys.exit(1)
+
+    # Set up logging
+    log_level = validated_args["--log-level"]
+    logging.basicConfig(
+        format="%(asctime)-15s %(levelname)s %(message)s", level=log_level.upper()
+    )
+
+    add_instance_states = DEFAULT_ADD_INSTANCE_STATES
+    postgres_db_name = DEFAULT_POSTGRES_DB_NAME
+    postgres_hostname = DEFAULT_POSTGRES_HOSTNAME
+    postgres_port = DEFAULT_POSTGRES_PORT
+    remove_instance_states = DEFAULT_REMOVE_INSTANCE_STATES
+
+    oneshot = validated_args["--oneshot"]
+    logging.debug("oneshot is %s.", oneshot)
+
+    postgres_password = validated_args["--postgres-password"]
+    if postgres_password is None:
+        with open(validated_args["--postgres-password-file"], "r") as file:
+            postgres_password = file.read()
+
+    postgres_username = validated_args["--postgres-username"]
+    if postgres_username is None:
+        with open(validated_args["--postgres-username-file"], "r") as file:
+            postgres_username = file.read()
+
+    rdp_password = validated_args["--rdp-password"]
+    if rdp_password is None:
+        with open(validated_args["--rdp-password-file"], "r") as file:
+            rdp_password = file.read()
+
+    rdp_username = validated_args["--rdp-username"]
+    if rdp_username is None:
+        with open(validated_args["--rdp-username-file"], "r") as file:
+            rdp_username = file.read()
+
+    vnc_password = validated_args["--vnc-password"]
+    if vnc_password is None:
+        with open(validated_args["--vnc-password-file"], "r") as file:
+            vnc_password = file.read()
+
+    vnc_username = validated_args["--vnc-username"]
+    if vnc_username is None:
+        with open(validated_args["--vnc-username-file"], "r") as file:
+            vnc_username = file.read()
+
+    private_ssh_key = validated_args["--private-ssh-key"]
+    if private_ssh_key is None:
+        with open(validated_args["--private-ssh-key-file"], "r") as file:
+            private_ssh_key = file.read()
+
+    db_connection_string = f"user={postgres_username} password={postgres_password} host={postgres_hostname} port={postgres_port} dbname={postgres_db_name}"
+    logging.debug("DB connection string is %s.", db_connection_string)
+
+    vpc_id = validated_args["--vpc-id"]
+
+    ec2 = boto3.resource("ec2", region_name="us-east-1")
```

Comment:
Should the region be an input parameter that defaults to `us-east-1`?  Or better yet, have it default to the region that the current instance is running in - something like what we do [here](
https://github.com/cisagov/ansible-role-freeipa-server/blob/ab1a00f2d3cadcd28373f86e05ccfe7aabd40e3c/files/00_setup_freeipa.sh#L154-L162) except we would use http://169.254.169.254/latest/meta-data/placement/region.

Reply:
Try on commit ec52ab4 for size.

Reply:
Love it!